terragrunt/modules/crates-io/compute-static/README.md

```diff
@@ -0,0 +1,23 @@
+# Compute@Edge Function for `static.crates.io`
+
+We are using a [Compute@Edge](https://docs.fastly.com/en/guides/compute-at-edge)
+function on [Fastly](https://fastly.com) to route incoming traffic for
+`static.crates.io` to S3. The function tries to get crates from the primary
+bucket, and will fail over to a fallback if it receives any HTTP `5xx`
+responses.
+
+## Deployment
+
+Build the function:
+
+```shell
+cd compute-static
+fastly compute build
+```
+
+Export an API token for Fastly and then run Terraform:
+
+```shell
+export FASTLY_API_KEY=""
+terraform plan
+```
```

Comment:
I'm worried that you have to remember to run `fastly compute build` to get the right configuration, and if you don't changes will be silently not applied. Can we use the [`external` data source](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/data_source) to invoke the compilation as part of planning?

Reply:
Done. I had to add a shell scripts that runs `fastly compute build` and returns a valid JSON object, though.